### ISSUE TITLE
Project level 5 

### ISSUE THREAD

**Issue by nanfang02** [nus-cs2030/2122-s2#397]
Hi i am a bit confused about how to approach level 5 of the project. It seems like i need to redo my entire level 3 events (arrive, serve, wait, done and leave). Or am i supposed to make the relevant changes at Simulate 5? 
Please send help thank you! :>

**Comment by Zhenyubbx** [comment]
Yep, you just need to make the relevant changes at Simulate 5. But if your constructors are changed, you will need to change your previous Simulate accordingly. Hope this helps!!!

**Comment by Guanjung27** [comment]
you wont really need to make alot of changes to other event if your implementation is correct, mainly your changes will be in simulate 5 , the customer class and Serve event.

**Comment by alwinngjw** [comment]
Hi, I think there should not be a need to change your events drastically if you implemented them correctly.

**Comment by Brianliew77** [comment]
One change u have to make is likely that u need to create a new constructor for SERVE event (depends on ur implementation tho), because previously all the serve times are 1.0, now it depends on the input. After that, update ur simulate4 to take in the new constructor of ServeEvent for simulate5. Hope this helps!

**Comment by kaitlyngoh** [comment]
you can change your customer contructor to take in a Lazy<Double>, so in the end youll only have to edit the customer and simulate5 class

**Comment by morksen123** [comment]
Dont need to redo just implement the Lazy & get() feature should do the trick

**Comment by bbelleeee** [comment]
i think dont need to redo!! you can try to replace the serve event with a new instance with the same attributes by getting it from the previous one except create a new customer that takes in the service time respectively
